bindings/protocols/modbus/ontology.ttl

```diff
@@ -229,6 +249,29 @@ binding:assignment rdf:type owl:AnnotationProperty .
           rdfs:comment "Function code identifies the operation requested by client, indicates to the server what kind of action to perform. For a normal response, the server simply echoes to the request the original function code."@en ;
           rdfs:label "Function"@en .
 
+        
+###  https://www.w3.org/2019/wot/modbus#PayloadDataType
+:PayloadDataType rdf:type owl:Class ;
+          owl:equivalentClass [ rdf:type owl:Class ;
+                                owl:oneOf ( xsd:integer
+                                            xsd:boolean
+                                            xsd:string
+                                            xsd:float
+                                            xsd:decimal
+                                            xsd:byte
+                                            xsd:short
+                                            xsd:int
+                                            xsd:long
+                                            xsd:unsignedByte
+                                            xsd:unsignedShort
+                                            xsd:unsignedInt
+                                            xsd:unsignedLong
+                                            xsd:double
+                                          )
+                              ] ;
+          rdfs:comment "Represents the data types that can be used to define the payload content in Modbus messages."@en ;
+          rdfs:label "PayloadDataType"@en .

```

Comment:
Do we need to model this as a class or can we also use a Dataproperty like I modelled above? I am not sure whether we can use the xsd types in the owl:oneOf. I can't find a reference which states that the xsd types are actually named individuals, otherwise it's not possible to use them with the owl:oneOf.

Reply:
I mainly decided to go with that design for consistency with the other classes (i.e., Entity and Function). This resulted in fewer changes in the renderer. The problem with the xsd types is that I can't find an ontology file that describes them and therefore I was kind of lost when choosing how to properly model them. Also modelling it as you are suggesting it means that the property we'll point to literals that are `xsd:integer` or `xsd:boolean` etc. (e.g. ` :b0 wot:PayloadDataType 3 . `) but we want to point the individuals defined by the URI `xsd:integer`. Shall we add them as individuals in our ontology to be more precise?

Reply:
That was also my problem, couldn't find the ontology of XSD, and only their XML schema. As you suggested, we could make individuals for them and then point to them.

Reply:
We could also make the `ResponseCode` class consistent with the other classes that you have defined like `Entity, Function` where you use a list pointing to named individuals with colon like `:Coil.` Here a URI pointing to the individual is used, but maybe there was a rational behind it. 
```
:ResponseCode rdf:type owl:Class ;
              owl:equivalentClass [ rdf:type owl:Class ;
                                    owl:oneOf ( <https://www.w3.org/2019/wot/modbus/errorCode#01>
                                                <https://www.w3.org/2019/wot/modbus/errorCode#02>
                                                <https://www.w3.org/2019/wot/modbus/errorCode#03>
                                                <https://www.w3.org/2019/wot/modbus/errorCode#04>
                                                <https://www.w3.org/2019/wot/modbus/errorCode#05>
                                                <https://www.w3.org/2019/wot/modbus/errorCode#06>
                                              )
                                  ] ;
              rdfs:subClassOf :Data_Field ;
              rdfs:comment "For an Exception Response, the server returns an exception code as data response field. Exception code defines the server condition that caused the exception as well as the reason of the error."@en ;
              rdfs:label "Exception code"@en .
```